source/extensions/filters/network/thrift_proxy/conn_manager.cc

```diff
@@ -217,50 +221,27 @@ bool ConnectionManager::ResponseDecoder::onData(Buffer::Instance& data) {
   return complete_;
 }
 
-FilterStatus ConnectionManager::ResponseDecoder::passthroughData(Buffer::Instance& data) {
-  passthrough_ = true;
-  return ProtocolConverter::passthroughData(data);
+FilterStatus ConnectionManager::ResponseDecoder::transportBegin(MessageMetadataSharedPtr metadata) {
+  return parent_.applyEncoderFilters(DecoderEvent::TransportBegin, metadata, protocol_converter_);
 }
 
-FilterStatus ConnectionManager::ResponseDecoder::messageBegin(MessageMetadataSharedPtr metadata) {
-  metadata_ = metadata;
-  metadata_->setSequenceId(parent_.original_sequence_id_);
+FilterStatus ConnectionManager::ResponseDecoder::transportEnd() {
+  ASSERT(metadata_ != nullptr);
 
-  if (metadata->hasReplyType()) {
-    success_ = metadata->replyType() == ReplyType::Success;
+  // Currently we don't support returning FilterStatus::StopIteration from encoder filters.
+  // Hence, this if-statement is always false.
+  if (parent_.applyEncoderFilters(DecoderEvent::TransportEnd, absl::any(), protocol_converter_) ==
```

Comment:
If we expect this to always be false we should probably add an assertion (even if it's handled).

Reply:
done